commons/impl/src/java/org/sakaiproject/commons/impl/PersistenceManagerImpl.java

```diff
@@ -247,6 +258,73 @@ public void run() {
         return sqlService.transact(transaction, "COMMONS_POST_DELETION_TRANSACTION");
     }
 
+    public boolean likePost(String postId, boolean toggle, String userId){
+        PostLike likeNow = getLike(postId, userId);
+        if(likeNow == null){    //if there is no existing Like, write a new one with the information provided
+            sqlService.dbWrite(POST_LIKE,new Object[]{userId, postId, toggle, new Timestamp(new Date().getTime())});
+            return true;
+        }
+        sqlService.dbWrite(LIKE_UPDATE, new Object[]{!likeNow.isLiked(), new Timestamp(new Date().getTime()), userId, postId});
+        return true;
+    }
+
+    public PostLike getLike(String postId, String userId){
+        List<PostLike> results = sqlService.dbRead(LIKE_GET, new Object[]{postId, userId}, new SqlReader<PostLike>(){
+                public PostLike readSqlResultRecord(ResultSet result) {
+                    return loadPostLikeFromResult(result);
+                }
+        });
+        if(results == null || results.size() == 0){
```

Comment:
```suggestion
        if(Collections.isEmpty(results)){
```

Reply:
Simplified as suggested on the latest commit.